### PR TITLE
SCMOD-5552: Added retries to autoscaler scaling

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**
https://autjira.microfocus.com/browse/SCMOD-5552

### CODE REVIEW

**Review by rado-straka — APPROVED** [review]

File: autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java [hunk 1]
```diff
@@ -29,11 +29,15 @@
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.Arrays;
+import java.util.logging.Level;
 import org.apache.http.client.utils.URIBuilder;
 import org.apache.http.message.BasicNameValuePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (dermot-hardy) [inline autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java:35]: We are adding both jul and slf4j logging imports?  Are we using two logging frameworks?

File: autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java [hunk 2]
```diff
@@ -51,22 +55,46 @@ private void patchInstances(final String appId, final int instances, final boole
...
+                    } else {
+                        LOG.error("Response code: " + response.getStatusLine().getStatusCode());
+                        LOG.error("Response Phrase: " + response.getStatusLine().getReasonPhrase());
+                        throw new ScalerException(response.getStatusLine().getReasonPhrase());
+                    }
+                }
+            } catch (final URISyntaxException | IOException ex) {
+                throw new ScalerException(String.format("Exception patching %s to %s instances.", appId, instances), ex);
+            } catch (final InterruptedException ex) {
+                throw new RuntimeException(ex);
```
Comment (dermot-hardy) [inline autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java]: You're catching the `InterruptedException` but not resetting the interrupt.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java (hunk 2 above)
Comment (Anthony-McGreevy) [reply autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java]: @dermot-hardy I have updated this PR

File: autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java [hunk 3]
```diff
@@ -51,22 +54,50 @@ private void patchInstances(final String appId, final int instances, final boole
...
+                        LOG.error("Response code: " + response.getStatusLine().getStatusCode());
+                        LOG.error("Response Phrase: " + response.getStatusLine().getReasonPhrase());
+                        throw new ScalerException(response.getStatusLine().getReasonPhrase());
+                    }
+                }
+            } catch (final URISyntaxException | IOException ex) {
+                throw new ScalerException(String.format("Exception patching %s to %s instances.", appId, instances), ex);
+            } catch (final InterruptedException ex) {
+                // Clear interupted flag
+                Thread.interrupted();
```
Comment (dermot-hardy) [inline autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java]: You should be setting the flag, not clearing it.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**